static/app/views/explore/savedQueries/savedQueriesTable.tsx

```diff
@@ -87,6 +88,27 @@ export function SavedQueriesTable({
   const {starQuery} = useStarQuery();
   const {updateQueryFromSavedQuery} = useSaveQuery();
 
+  const [starredIds, setStarredIds] = useState<number[]>([]);
+
+  // Initialize starredIds state when queries have been fetched
+  useEffect(() => {
+    if (isFetched === true) {
+      setStarredIds(data?.filter(row => row.starred).map(row => row.id) ?? []);
+    }
+  }, [isFetched, data]);
+
+  const starQueryHandler = useCallback(
+    (id: number, starred: boolean) => {
+      starQuery(id, starred);
```

Comment:
If the starred query request fails for any reason, we should revert the `starredIds` state by wrapping this in a try/catch

Reply:
good point!